File: Go/Two Pointers/Two-Sum-II.go
package main

func main() {

}

func twoSum(numbers []int, target int) []int {
	left, right := 0, len(numbers)-1

	for {
		sum := numbers[left] + numbers[right]

		if sum < target {
			left++
		} else if sum > target {
			right--
		} else {
			return []int{left + 1, right + 1}
		}
	}
}

File: Rust/Two Pointers/Two-Sum-II.rs
struct Solution;

use std::cmp::Ordering;

impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let (mut left, mut right) = (0, numbers.len() - 1);
        loop {
            match (numbers[left] + numbers[right]).cmp(&target) {
                Ordering::Equal => return vec![(left + 1) as i32, (right + 1) as i32],
                Ordering::Greater => right -= 1,
                Ordering::Less => left += 1,
            }
        }
    }
}

fn main() {}
